src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -156,10 +157,22 @@ private function registerReceivers(ContainerBuilder $container)
         $receiverMapping = array();
         foreach ($container->findTaggedServiceIds('messenger.receiver') as $id => $tags) {
             foreach ($tags as $tag) {
-                $receiverMapping[$tag['id'] ?? $id] = new Reference($id);
+                $receiverMapping[$tag['name'] ?? $id] = new Reference($id);
             }
         }
 
         $container->getDefinition('messenger.receiver_locator')->replaceArgument(0, $receiverMapping);
     }
+
+    private function registerSenders(ContainerBuilder $container)
+    {
+        $senderLocatorMapping = array();
+        foreach ($container->findTaggedServiceIds('messenger.sender') as $id => $tags) {
+            foreach ($tags as $tag) {
+                $senderLocatorMapping[$tag['name'] ?? $id] = new Reference($id);
```

Comment:
Shouldn't it be available by both tag name and service id?

Reply:
I don't think so actually. If it has a `name`, then that's what you use. It's aligned with what we discussed on other PRs regarding a single way of configuring/using the adapters.

Reply:
not allowing by-id forces knowing names, when knowing ids is easier sometimes (`debug:container --tag messenger.sender`)

Reply:
Alright, you are two thinking that way and I'm mixed. Added, thanks 👍